runner/runner.go

```diff
@@ -263,6 +264,27 @@ type Runner struct {
 
 	providers   map[string]common.Provider
 	credentials map[string]config.Github
+
+	controllerInfo params.ControllerInfo
+}
+
+func (r *Runner) GetControllerInfo(ctx context.Context) (params.ControllerInfo, error) {
+	if r.controllerInfo == (params.ControllerInfo{}) {
```

Comment:
Everything else looks fine!

Reply:
Ahh. My mistake. ```os.Hostname()``` can return an error. So that would be something like:

```go
hostname, err := os.Hostname()
if err != nil {
    return params.ControllerInfo{}, errors.Wrap(err, "getting hostname")
}
return params.ControllerInfo{
  ControllerID: r.poolManagerCtrl.controllerID,
  Hostname: hostname,
}, nil
```

Reply:
hm .. accessing the id through Runner.poolManagerCtrl.controllerID is not that easy .. as you decoupled the access via an interface, I didn't want to extend this .. and type assertion seems not the right way .. 

os.Hostname will probably not really error .. and if so we will ignore it anyway .. we cannot crash, right? 
So I went for having no error result, but there's still the context in the runner.GetControllerInfo .. do you think that still makes  much sense ?

Reply:
You are right. Then it might be worth adding a new un-exported field to ```Runner{}``` like ```controllerID uuid.UUID```, and cache the controller ID there for later use. 

Note, the ```NewRunner()``` function is only called once. If you want to catch a hostname change, you probably want to call that in ```GetControllerInfo()```. The call to ```os.Hostname()``` shouldn't error, but if they added that value as a return, it *may* happen, in which case you risk getting an empty string there. Is that ok in your current code path?

Reply:
> So I went for having no error result, but there's still the context in the runner.GetControllerInfo .. do you think that still makes much sense ?

The context is useful if we want to limit who gets to fetch this info. There is no real sensitive info returned by this function, but we should probably still check for authorization here. In the future we may extend this function to return more metadata about the controller, especially if we will decouple the code and possibly deploy multiple instances of that particular component.

Reply:
I will not nitpick too much on this. I am okay with merging as is, and making changes later as needed. The only thing I think is worth changing is the location of the call to ```os.Hostname()```. It's worth moving it into ```GetControllerInfo()``` as that should resolve the hostname on every call to ```GetControllerInfo()```, as opposed to caching it for the duration of the lifetime of the current process.

You mentioned previously that you would rather pick up on hostname changes without restarting garm.

Reply:
> You mentioned previously that you would rather pick up on hostname changes without restarting garm.

right .. thanks for your reviews and your patience with me :smile:

Reply:
Thank you for the work you put into this!